### ISSUE TITLE
Google Maps -- GPS won't work unless phone has fix prior to launching gmaps

### ISSUE THREAD

**Issue by ASerbinski** [GrapheneOS/os-issue-tracker#1856]
Specific details: Pixel 6/7 pro, google services are NOT installed.

Issue: if Google Maps is launched before a GPS fix has been acquired, location in gmaps is never updated (grey dot at last position). If a fix has been acquired before launching gmaps (I.e. gpstest, wait until fix, then launch gmaps), then location works correctly (blue dot, tracks location).

**Comment by W1zardK1ng** [comment]
This happens for me also on my Pixel 6a, the workaround is to press on the current location icon and minimize the app and reopen it. once the app is reopened the current location will be updated .

**Comment by thestinger** [comment]
Is this still an issue for you?

**Comment by lbdroid** [comment]
@thestinger ; Yes it is.

**Comment by ASerbinski** [comment]
Sorry, that was my alter ego speaking.

**Comment by invertedEcho** [comment]
I have the same issue, I'm on a Pixel 6a. I have to open gpstest before opening GMaps, but my location doesn't refresh.

**Comment by Zeit08** [comment]
Same issue on latest release, Pixel 6A, no play services. Is it a problem due to missing play services and hence not a bug of GrapheneOS?

**Comment by invertedEcho** [comment]
@Zeit08 
I have sandboxed Play Services installed, so I dont think so.
@thestinger Any update on this? Its quite annoying not to have GPS working

**Comment by thestinger** [comment]
Please don't post on the issue without having any new information to add and in particular please don't ping me.

**Comment by gsture** [comment]
I have the same issue. It's why I have a seperated phone for navigation because it doesn't work reliably on grapheneos. It's not only maps though, I regularly have to exit (minimizing does not help) applications and reopening before they can get a location.

**Comment by thestinger** [comment]
We're unable to replicate any issues.

**Comment by gsture** [comment]
In my case the apps that are having these issues are installed in a work profile. Have you tried that?

**Comment by thestinger** [comment]
That's different what was was filed here. Work profiles are known to have issues including on the stock OS with foreground permissions. We recommend user profiles.

**Comment by ASerbinski** [comment]
@thestinger From a freshly wiped system, install just gmaps (and gpstest) either from apk or from fdroid+aurora store into owner profile. Nothing else. Set up gmaps by skipping the warning about gplay and using no google account. Then kill gmaps and let the phone sit for an hour so that any gps data goes stale.

Now wake up the phone and launch gmaps with clear view of sky and wait for long enough to get a GPS fix. Notice that it doesn't. Now switch to gpstest and notice that it get a fix. Switch back to gmaps and still no fix. Kill gmaps and relaunch, now it will get a fix.

@thestinger ; I understand that you're from Toronto. If need be, we could set up a time where I could help you replicate the issue.

**Comment by thestinger** [comment]
That sounds like a Google Maps bug with their support for non-Play location since it doesn't involve the parts GrapheneOS adds for Google Play compatibility.

**Comment by ASerbinski** [comment]
Yes, very likely since from what I can tell, it only affects gmaps. Nevertheless, I can't replicate it on other non-gplay aosp builds on other devices, such as lineageos on nexus 6 or aosp on dragonboard 820c (although to be fair, I wrote the GPS hal for the db820c myself so that it fakes the first lock in order to allow nav destination to be input prior to location lock).

**Comment by thestinger** [comment]
It's only relevant if you're testing on Android 13 everywhere. Are you?

**Comment by waynesworld937** [comment]
Same issue on my Pixel 7 Pro. I have to close the app, reopen and create a nav quickly. If I don't use it straight away, and go to another app, it loses lock and I have to start again.

**Comment by ASerbinski** [comment]
@thestinger ; is a fix for this expected to be incorporated into the next build? Current build still exhibits this issue.

**Comment by ASerbinski** [comment]
Well there's been no apparent change at all, it remains exactly as I described when I created this issue. I just tested it again a minute ago to verify.

**Comment by muhomorr** [comment]
@ASerbinski Do you still use Google Maps app without Play services and its dependencies?

**Comment by ASerbinski** [comment]
@muhomorr - you probably should create a new thread on that topic.

```
version 11.111.0202

com.google.android.apps.maps
versionCode 1067348547

targetSdk 34
minSdk 23

Installed: Feb 14, 2024; 8:15 PM
```

**Comment by ASerbinski** [comment]
@muhomorr -- the entire point of this thread is exactly NO GSERVICES.
